### PR TITLE
imp: add logic for setting `NextSeqRecv`, `NextSeqAck` in `OnChanUpgradeOpen`

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

[ref spec changes ](https://github.com/cosmos/ibc/issues/1061)

note that the counterparty last send seq and other pruning related topics will be addressed in upcoming PRs.
this PR covers just the logic for setting `NextSeqRecv` and `NextSeqAck` in the upgrade handlers.

### Commit Message / Changelog Entry

```text
type: commit message
```

see the [guidelines](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) for commit messages. (view raw markdown for examples)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/core/04-channel/keeper/packet_test.go
Comment (charleenfei) [inline modules/core/04-channel/keeper/packet_test.go:354]: i removed these tests bc we aren't checking these sequences in `RecvPacket` anymore s

File: modules/core/04-channel/keeper/upgrade.go
Comment (AdityaSripal) [inline modules/core/04-channel/keeper/upgrade.go:438]: This should also happen on Ack

File: modules/core/04-channel/keeper/upgrade.go
Comment (charleenfei) [inline modules/core/04-channel/keeper/upgrade.go:558]: pruning seq can be set here, in any upgrade case, so that we can get the `counterpartyUpgrade.NextSequenceSend` before it gets wiped

File: modules/core/04-channel/keeper/upgrade_test.go
Comment (AdityaSripal) [inline modules/core/04-channel/keeper/upgrade_test.go:1163]: I don't understand the reset comment

Comment (AdityaSripal) [reply modules/core/04-channel/keeper/upgrade_test.go:1163]: Should we also be assert the counterpartyNextSequenceSend is 2 here?

File: modules/core/04-channel/keeper/upgrade_test.go
Comment (AdityaSripal) [inline modules/core/04-channel/keeper/upgrade_test.go:1295]: Should we also be assert the counterpartyNextSequenceSend is 2 here?

**Review by AdityaSripal — APPROVED** [review]
Only a comment on even more assertion in test, but its implicitly checked. Should be merged, with an optional isssue opened to add this if people find it important. I'm personally ambivalent, just saw that the test comment may be expecting that to be the case

File: modules/core/04-channel/keeper/upgrade_test.go
Comment (charleenfei) [reply modules/core/04-channel/keeper/upgrade_test.go:1295]: that set isnt in this PR, it's in [this one ](https://github.com/cosmos/ibc-go/pull/5460/files), the test can be updated there tho for sure!

File: modules/core/04-channel/keeper/upgrade_test.go
Comment (charleenfei) [reply modules/core/04-channel/keeper/upgrade_test.go:1163]: i believe we set the sequences to 1 so that the upgrade proof can be verified without the exact last sequence send

**Review by crodriguezvega — APPROVED** [review]
Fantastic, @charleenfei! Thank you for adapting the code so quickly to the spec updates.

File: modules/core/04-channel/keeper/upgrade_test.go
Comment (charleenfei) [reply modules/core/04-channel/keeper/upgrade_test.go:1163]: You're absolutely correct! Will update.

File: modules/core/04-channel/keeper/upgrade_test.go
Comment (damiannolan) [reply modules/core/04-channel/keeper/upgrade_test.go:1093]: Agree it may be easier to parse if we kept this test func and added a separate one for testing the ordering changes 👍🏻

**Review by damiannolan — APPROVED** [review]
Nice work! Agree with @DimitrisJim's testing comments for improvements :) 

LGTM

File: modules/core/04-channel/keeper/upgrade_test.go
Comment (charleenfei) [reply modules/core/04-channel/keeper/upgrade_test.go:1093]: would you prefer for me to fix that in this pr? or merge and open another pr in the future for this testing refactor?

Comment (charleenfei) [reply modules/core/04-channel/keeper/upgrade_test.go:1356]: i'll fix it in this pr since still waiting for others to be merged!

**Review by colin-axner — APPROVED** [review]

**Comment by charleenfei** [comment]
i'm gonna merge this one first after checks pass before [this one](https://github.com/cosmos/ibc-go/pull/5469), as there will be some reworking of that logic

**Pull request merged**
